src/test/unit/common/languageserver/languageserver.test.ts

```diff
@@ -0,0 +1,153 @@
+        get: () => Promise.resolve(),
+        delete: () => Promise.resolve(),
+      },
+    } as unknown as ExtensionContext;
+
+    SecretStorageAdapter.init(extensionContext);
+    configuration = {
+      getCustomCliPath(): string | undefined {
+        return 'testPath';
+      },
+      getToken(): Promise<string | undefined> {
+        return Promise.resolve('testToken');
+      },
+      shouldReportEvents: true,
+      shouldReportErrors: true,
+      getPreviewFeatures(): PreviewFeatures {
+        return {
+          advisor: false,
+          reportFalsePositives: false,
+          lsAuthenticate: true,
+        };
+      },
+      getSnykLanguageServerPath(): string {
+        return 'testPath';
+      },
+    } as unknown as IConfiguration;
+  });
+
+  teardown(() => {
+    sinon.restore();
+  });
+
+  test('LanguageServer should provide correct initialization options', async () => {
+    languageServer = new LanguageServer(
+      extensionContext,
+      configuration,
+      {} as unknown as ILanguageClientAdapter,
+      {} as unknown as IVSCodeWorkspace,
+    );
+    const expectedInitializationOptions = {
+      activateSnykCode: 'false',
+      activateSnykOpenSource: 'false',
+      activateSnykIac: 'false',
+      token: 'testToken',
+      cliPath: 'testPath',
+      enableTelemetry: 'true',
+      sendErrorReports: 'true',
+      integrationName: 'VS_CODE',
+      integrationVersion: '0.0.0',
+    };
+
+    deepStrictEqual(await languageServer.getInitializationOptions(), expectedInitializationOptions);
+  });
+
+  test('LanguageServer starts language client if preview enabled', async () => {
+    const lc = sinon.spy({
+      start(): Promise<void> {
+        return Promise.resolve();
+      },
+    } as unknown as LanguageClient);
+
+    const lca = sinon.spy({
+      create(): LanguageClient {
+        return lc as unknown as LanguageClient;
+      },
+    });
+
+    languageServer = new LanguageServer(
+      extensionContext,
+      configuration,
+      lca as unknown as ILanguageClientAdapter,
+      stubWorkspaceConfiguration('http.proxy', undefined),
+    );
+    await languageServer.start();
+    sinon.assert.called(lca.create);
+    sinon.assert.called(lc.start);
+    sinon.verify();
+  });
+
+  test('LanguageServer does not start language client if preview not enabled', async () => {
+    const lca = sinon.stub();
+    sinon.stub(configuration, 'getPreviewFeatures').returns({
+      advisor: false,
+      reportFalsePositives: false,
+      lsAuthenticate: false,
+    });
+
+    languageServer = new LanguageServer(
+      extensionContext,
+      configuration,
+      lca as unknown as ILanguageClientAdapter,
+      stubWorkspaceConfiguration('http.proxy', undefined),
+    );
+    await languageServer.start();
+    sinon.assert.notCalled(lca);
+    sinon.verify();
+  });
+
+  test('LanguageServer adds proxy settings to env of started binary', async () => {
```

Comment:
You don't fill proxy environment settings - is it intended? The test case name gave this expectation.

Reply:
Yep, don't need it filled, just need to check the variable is there :)